client/icecc-create-env.in

```diff
@@ -81,10 +81,14 @@ resolve_path()
 # This could use realpath, but that's reportedly not that widely available.
 convert_path_cdup ()
 {
-  local filename="$1"
-  local directory=$(dirname $filename)
-  local fixed_directory=$(cd "$directory" >/dev/null && pwd)
-  echo ${fixed_directory}/$(basename $filename)
+  # Remove all /./ sequences.
+  local path=${1//\/.\//\/}
+
+  # Remove dir/.. sequences.
+  while [[ $path =~ ([^/][^/]*/\.\./) ]]; do
+    path=${path/${BASH_REMATCH[0]}/}
+  done
+  echo $path
```

Comment:
Why is this part necessary?

Reply:
I tested some cases where one might want to remap to a path in the compiler package that doesn't exist on the local system, and therefore the pwd approach would not work. This might be over-reaching now that I look at it again, because I can't think of legitimate reason to do this. Ideally you are trying to make the compiler package match the local system in such a case.

Reply:
Hm, I would prefer to not be doing this with Bash regexes if possible. They've always been a little odd in the way they work. I think we can just detect that the path doesn't exist here and therefore fail and require an absolute path? Would that be a good option, do you think?

Reply:
@ossilator True enough - the argument I was making was more about future readers than the writer. I've used Bash regexes many times before with success, but they tend to confuse people reading them in the future who aren't familiar with them.

In any case, I think overall I would prefer we use a different mechanism for adding paths that don't exist locally anyway and prefer that we use absolute paths with it unless we have a good reason not to.